Follow-up fix to 277527@main

csaavedra opened the pull request:
#### 93c46cf87c69e32f3e0c38bdfbada8adc0fd8ab6
<pre>
Follow-up fix to 277527@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=272742">https://bugs.webkit.org/show_bug.cgi?id=272742</a>

Reviewed by Alex Christensen and Don Olmstead.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.mm:
(IPC::ArgumentCoder&lt;WebKit::RemoteLayerBackingStore&gt;::encode):
* Source/WebKit/Shared/mac/ObjCObjectGraph.mm:
(WebKit::ObjCObjectGraph::decode):
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPCSemaphore::JSIPCSemaphore):
(WebKit::IPCTestingAPI::JSIPCConnectionHandle::JSIPCConnectionHandle):
(WebKit::IPCTestingAPI::JSIPCConnection::JSIPCConnection):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::JSIPCStreamClientConnection):
(WebKit::IPCTestingAPI::JSIPCStreamServerConnectionHandle::JSIPCStreamServerConnectionHandle):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::JSIPCStreamConnectionBuffer):
(WebKit::IPCTestingAPI::JSSharedMemory::JSSharedMemory):
(WebKit::IPCTestingAPI::JSIPC::JSIPC):
(WebKit::IPCTestingAPI::extractSyncIPCMessageInfo):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::streamBuffer):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::setSemaphores):
(WebKit::IPCTestingAPI::extractIPCStreamMessageInfo):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::sendWithAsyncReply):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::sendSyncMessage):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::sendIPCStreamTesterSyncCrashOnZero):
(WebKit::IPCTestingAPI::JSIPCSemaphore::signal):
(WebKit::IPCTestingAPI::JSIPCSemaphore::waitFor):
(WebKit::IPCTestingAPI::JSSharedMemory::readBytes):
(WebKit::IPCTestingAPI::arrayBufferSpanFromValueRef):
(WebKit::IPCTestingAPI::JSSharedMemory::writeBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::readHeaderBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::readDataBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::readBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::writeHeaderBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::writeDataBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::writeBytes):
(WebKit::IPCTestingAPI::JSIPC::processTargetFromArgument):
(WebKit::IPCTestingAPI::JSIPC::addMessageListener):
(WebKit::IPCTestingAPI::messageNameFromArgument):
(WebKit::IPCTestingAPI::encodeTypedArray):
(WebKit::IPCTestingAPI::encodeRemoteRenderingBackendCreationParameters):
(WebKit::IPCTestingAPI::encodeSharedMemory):
(WebKit::IPCTestingAPI::encodeStreamConnectionBuffer):
(WebKit::IPCTestingAPI::encodeStreamServerConnectionHandle):
(WebKit::IPCTestingAPI::encodeSemaphore):
(WebKit::IPCTestingAPI::encodeConnectionHandle):
(IPC::ArgumentCoder&lt;WebKit::IPCTestingAPI::VectorEncodeHelper&gt;::encode):
(WebKit::IPCTestingAPI::JSMessageListener::jsDescriptionFromDecoder):
* Tools/TestWebKitAPI/Tests/IPC/ArgumentCoderTests.cpp:
(TestWebKitAPI::ArgumentCoderEncoderDecoderTest&lt;IPC::StreamConnectionEncoder&gt;::Impl::Impl):
(TestWebKitAPI::EncodingCounter::CounterValues::CounterValues):
(TestWebKitAPI::EncodingCounter::EncodingCounter):
(IPC::ArgumentCoder&lt;TestWebKitAPI::EncodingCounter&gt;::encode):
(IPC::ArgumentCoder&lt;TestWebKitAPI::DecodingMoveCounter&gt;::encode):
(IPC::ArgumentCoder&lt;TestWebKitAPI::DecodingMoveCounter&gt;::decode):
* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(IPC::ArgumentCoder&lt;CFHolderForTesting&gt;::encode):
(IPC::ArgumentCoder&lt;CFHolderForTesting&gt;::decode):
(IPC::ArgumentCoder&lt;ObjCHolderForTesting&gt;::encode):
(IPC::ArgumentCoder&lt;ObjCHolderForTesting&gt;::decode):

Canonical link: <a href="https://commits.webkit.org/277916@main">https://commits.webkit.org/277916@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14daeeacd8dcaa7b6c225568e5ba9636643b1819

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48906 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28119 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51873 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51593 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44976 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51211 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34071 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25647 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39977 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/49536 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25768 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42171 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21082 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23228 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43343 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6961 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/42205 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45159 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43844 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53504 "Built successfully") | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/48397 "Found 1 new JSC stress test failure: stress/scoped-arguments-table-should-be-tolerant-for-oom.js.bytecode-cache (failure)") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23957 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20221 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47287 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25220 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42380 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46241 "Found 4 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/selection/listbox, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/selection/menulist, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/event-listener, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/value/basic (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26029 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/55892 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6997 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24940 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11505 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

csaavedra commented on Source/WebKit/Shared/mac/ObjCObjectGraph.mm:
```diff
@@ -189,11 +189,11 @@ static bool shouldTransformGraph(id object, const ObjCObjectGraph::Transformer&
         return false;
 
-    switch (type) {
+    switch (*type) {
```
I landed a fix for this earlier today so you won't need it.

achristensen07 replied:
This change is indeed still needed.

csaavedra replied:
How come? https://github.com/WebKit/WebKit/commit/3ef82afb51cba7366ed3882a8c24da37ba8031b7

achristensen07 replied:
Oh you're right.  Rebasing shows you've already done this exact change.

donny-dont commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1794,5 +1794,5 @@
 // FIXME: Other platforms should finish migration to *.serialization.in.
 #if PLATFORM(COCOA)
-#define ONLY_MODERN_SERIALIZATION 1
+#define HAVE_ONLY_MODERN_SERIALIZATION 1
```
What's the state of GTK/WPE with this turned on by default? I'm able to compile 🪟 s with this `ON`.

csaavedra replied:
Not ready yet, but I'm working on it.

csaavedra replied:
Now it's ready, so I submitted https://github.com/WebKit/WebKit/pull/27734 . Not sure if this is used elsewhere so I marked it as a draft for now.

achristensen07 committed:
Follow-up fix to 277527@main
https://bugs.webkit.org/show_bug.cgi?id=272742

Reviewed by Alex Christensen and Don Olmstead.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreeTransaction.mm:
(IPC::ArgumentCoder<WebKit::RemoteLayerBackingStore>::encode):
* Source/WebKit/Shared/mac/ObjCObjectGraph.mm:
(WebKit::ObjCObjectGraph::decode):
* Source/WebKit/WebProcess/WebPage/IPCTestingAPI.cpp:
(WebKit::IPCTestingAPI::JSIPCSemaphore::JSIPCSemaphore):
(WebKit::IPCTestingAPI::JSIPCConnectionHandle::JSIPCConnectionHandle):
(WebKit::IPCTestingAPI::JSIPCConnection::JSIPCConnection):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::JSIPCStreamClientConnection):
(WebKit::IPCTestingAPI::JSIPCStreamServerConnectionHandle::JSIPCStreamServerConnectionHandle):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::JSIPCStreamConnectionBuffer):
(WebKit::IPCTestingAPI::JSSharedMemory::JSSharedMemory):
(WebKit::IPCTestingAPI::JSIPC::JSIPC):
(WebKit::IPCTestingAPI::extractSyncIPCMessageInfo):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::streamBuffer):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::setSemaphores):
(WebKit::IPCTestingAPI::extractIPCStreamMessageInfo):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::sendWithAsyncReply):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::sendSyncMessage):
(WebKit::IPCTestingAPI::JSIPCStreamClientConnection::sendIPCStreamTesterSyncCrashOnZero):
(WebKit::IPCTestingAPI::JSIPCSemaphore::signal):
(WebKit::IPCTestingAPI::JSIPCSemaphore::waitFor):
(WebKit::IPCTestingAPI::JSSharedMemory::readBytes):
(WebKit::IPCTestingAPI::arrayBufferSpanFromValueRef):
(WebKit::IPCTestingAPI::JSSharedMemory::writeBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::readHeaderBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::readDataBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::readBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::writeHeaderBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::writeDataBytes):
(WebKit::IPCTestingAPI::JSIPCStreamConnectionBuffer::writeBytes):
(WebKit::IPCTestingAPI::JSIPC::processTargetFromArgument):
(WebKit::IPCTestingAPI::JSIPC::addMessageListener):
(WebKit::IPCTestingAPI::messageNameFromArgument):
(WebKit::IPCTestingAPI::encodeTypedArray):
(WebKit::IPCTestingAPI::encodeRemoteRenderingBackendCreationParameters):
(WebKit::IPCTestingAPI::encodeSharedMemory):
(WebKit::IPCTestingAPI::encodeStreamConnectionBuffer):
(WebKit::IPCTestingAPI::encodeStreamServerConnectionHandle):
(WebKit::IPCTestingAPI::encodeSemaphore):
(WebKit::IPCTestingAPI::encodeConnectionHandle):
(IPC::ArgumentCoder<WebKit::IPCTestingAPI::VectorEncodeHelper>::encode):
(WebKit::IPCTestingAPI::JSMessageListener::jsDescriptionFromDecoder):
* Tools/TestWebKitAPI/Tests/IPC/ArgumentCoderTests.cpp:
(TestWebKitAPI::ArgumentCoderEncoderDecoderTest<IPC::StreamConnectionEncoder>::Impl::Impl):
(TestWebKitAPI::EncodingCounter::CounterValues::CounterValues):
(TestWebKitAPI::EncodingCounter::EncodingCounter):
(IPC::ArgumentCoder<TestWebKitAPI::EncodingCounter>::encode):
(IPC::ArgumentCoder<TestWebKitAPI::DecodingMoveCounter>::encode):
(IPC::ArgumentCoder<TestWebKitAPI::DecodingMoveCounter>::decode):
* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(IPC::ArgumentCoder<CFHolderForTesting>::encode):
(IPC::ArgumentCoder<CFHolderForTesting>::decode):
(IPC::ArgumentCoder<ObjCHolderForTesting>::encode):
(IPC::ArgumentCoder<ObjCHolderForTesting>::decode):

Canonical link: https://commits.webkit.org/277916@main
